cpp/include/cudf/ast/expressions.hpp

```diff
@@ -474,6 +502,48 @@ class operation : public expression {
   std::vector<std::reference_wrapper<expression const>> const operands;
 };
 
+/**
+ * @brief A expression referring to data from a column in a table.
+ */
+class column_name_reference : public expression {
```

Comment:
column names are not available before reading the parquet file (metadata) before calling the reader. That's why this is added.

Reply:
Can we fetch metadata from the file first, before reading the rest of the file? I am very hesitant to introduce this logic into the AST feature. It is not well suited to the rest of libcudf which is generally unaware of column names.

Reply:
Another alternative would be to create an arbitrary mapping of input column names to integers (assign the first name `0`, second name `1`, and increment from there). Then we could reindex the column references at a later time once the true indices for those column names are known.

Reply:
I thought about that too. But this would leave the user with overhead of managing this ordering. Column name reference seems easier to construct for the user.
Besides, they can still do that with the existing code, by using column_reference.

Reply:
We discussed this offline. We want to provide users with a friendly API (using string names). Unfortunately that introduces some complexity to the visitor pattern, since AST filtering supports names but libcudf (and the AST transform) do not support names. We're settling for some comments explaining the conflicting goals, since there wasn't an obvious way to make this better (balancing code duplication, user friendliness, etc.).